[supply] adding multiple aabs

leandroBorgesFerreira opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
fixes #13536. Add the possibility to pass multiple aab just like apk_paths. 

### Description
The code has the same logic as apk_path command. 
As a last test, I uploaded an array of aab to Play Store using the `aab_paths` without any problems.

leandroBorgesFerreira committed:
adding multiple aabs

leandroBorgesFerreira commented:
I signed it!

leandroBorgesFerreira committed:
fixing identation

bdrelling reviewed (commented):
Just a few tiny things that might be worth addressing.

leandroBorgesFerreira committed:
pr fixes

leandroBorgesFerreira commented on supply/lib/supply/options.rb:
```diff
@@ -128,6 +128,20 @@ def self.available_options
                                      end),
+        FastlaneCore::ConfigItem.new(key: :aab_paths,
+                                     env_name: "SUPPLY_AAB_PATHS",
+                                     conflicting_options: [:apk_path, :apk_paths, :aab],
```
I think we should address it in this PR. Good catch.

leandroBorgesFerreira commented:
@bdrelling I just addressed your comments. The pipeline didn't run because of some infrastructure problem in Circle CI. would you mind retriggering it?

joshdholtz commented on supply/lib/supply/uploader.rb:
```diff
@@ -152,11 +152,17 @@ def upload_mapping(apk_version_codes)
-      aab_path = Supply.config[:aab]
-      return [] unless aab_path
+      aab_paths = [Supply.config[:aab]] unless (aab_paths = Supply.config[:aab_paths])
+      # return [] unless aab_paths
```
@leandroBorgesFerreira Can we remove this commented out line? 🙃

leandroBorgesFerreira replied:
fixed =]

joshdholtz reviewed (changes requested):
This looks great! Thanks for making this change 😊 Can you remove that one commented out line and then we can get this merged and released today 💪

leandroBorgesFerreira committed:
removing unwanted comments

joshdholtz commented on supply/lib/supply/uploader.rb:
```diff
@@ -152,11 +152,17 @@ def upload_mapping(apk_version_codes)
     def upload_bundles
-      aab_path = Supply.config[:aab]
-      return [] unless aab_path
+      aab_paths = [Supply.config[:aab]] unless (aab_paths = Supply.config[:aab_paths])
```
Ah, one more thing! There is a very small possibility that `Supply.config[:aab]` might be `nil` so `aab_paths` could be `[nil]`. Can we add a `aab_paths.compact!` after this line? Thhheeennnn I think we should be good 💪

leandroBorgesFerreira replied:
wouldn't be better to use `compact` instead of `compact!`? `compact!` would return nil if no removal were made.

joshdholtz reviewed (changes requested):
One more quick change!

leandroBorgesFerreira committed:
removing nil from aab_paths variable

joshdholtz commented on supply/lib/supply/uploader.rb:
```diff
@@ -152,11 +152,17 @@ def upload_mapping(apk_version_codes)
-      aab_path = Supply.config[:aab]
-      return [] unless aab_path
+      aab_paths = [Supply.config[:aab]] unless (aab_paths = Supply.config[:aab_paths])
+      return [] unless aab_paths.compact
```
Ah sorry, I was looking at how this was implemented in the `upload_apks` - https://github.com/fastlane/fastlane/pull/13543/files#diff-fd047a79b6e303ba583ff34a4c489d78L134. `upload_apks` doesn't early return so doing a `aab_paths.compact!` on its own lined would worked.

I think it will always get passed with `return [] unless aab_paths.compact` (as long as `aab_paths` isn't `nil`). If `aab_paths` is `nil` this looks like it will 💥 .

I think what we actually want is something like...

```suggestion
      return [] unless aab_paths
      aab_paths.compact!
```

This way it will early return with an empty array if `aab_paths` is `nil`, otherwise it will remove all possible `nil` files inside of `aab_paths`

joshdholtz reviewed (changes requested):
One small suggestion and then I think we are 💯

Commit:
PR fix

Co-Authored-By: leandroBorgesFerreira <lehen01@gmail.com>

leandroBorgesFerreira commented:
So clicking in "apply suggestion" wasn't a great idea hahaha. Now this PR contains commits that were not done by me... Anyway, I just created another PR, just like this one, but it passes the `cla/google` check. https://github.com/fastlane/fastlane/pull/13639

joshdholtz commented:
@leandroBorgesFerreira Yeah, that is expected from googlebot - its okay 😊 Its not more of a warning than an error 😉 We can still continue with this PR

joshdholtz reviewed (approved):
Thanks for this contribution ❤️ Looking forward to getting this one released 🚀
